src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -69,6 +76,20 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService,
         return ResponseEntity.ok(result);
     }
 
+    /**
+     * Given an exerciseId and a submissionId, the method retrieves from the database all the data needed by the tutor
+     * to assess the submission.
+     *
+     * If the tutor has already started assessing the submission, then we also return all the results the tutor has
+     * already inserted.
+     *
+     * If another tutor has already started working on this submission, the system tries to find another one without
+     * any result, and if it finds any, return it
+     *
+     * @param exerciseId the id of the exercise we want the submission
+     * @param submissionId the id of the submission we want
+     * @return a Participation of the tutor in the submission
+     */
     @GetMapping("/exercise/{exerciseId}/submission/{submissionId}")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<Participation> getDataForTutor(@PathVariable Long exerciseId, @PathVariable Long submissionId) {
```

Comment:
This is a bad method call, because looking at the REST Call, I would assume to retrieve a submission here. Please think about at least renaming the method name, when not only changing the REST API endpoint

Reply:
One possible method name would be "selectNextSubmissionToBeAssessed(...)"

Reply:
Added a todo